crates/red_knot_test/src/parser.rs

```diff
@@ -143,12 +257,45 @@ struct EmbeddedFileId;
 #[derive(Debug)]
 pub(crate) struct EmbeddedFile<'s> {
     section: SectionId,
-    pub(crate) path: String,
+    path: EmbeddedFilePath<'s>,
     pub(crate) lang: &'s str,
-    pub(crate) code: &'s str,
+    code_blocks: Vec<CodeBlock<'s>>,
+}
 
-    /// The offset of the backticks beginning the code block within the markdown file
-    pub(crate) backtick_offset: TextSize,
+impl EmbeddedFile<'_> {
+    /// Returns the full code for the embedded file, which can consist
+    /// of multiple code blocks.
+    pub(crate) fn code(&self) -> Cow<'_, str> {
+        if self.code_blocks.len() == 1 {
+            Cow::Borrowed(self.code_blocks[0].code)
+        } else {
+            let mut merged_code = String::new();
+            for block in &self.code_blocks {
+                // Treat empty code blocks as non-existent, instead of creating
+                // an additional empty line:
+                if block.code.is_empty() {
+                    continue;
+                }
+
+                if !merged_code.is_empty() {
+                    merged_code.push('\n');
+                }
+                merged_code.push_str(block.code);
+            }
+            Cow::Owned(merged_code)
+        }
+    }
```

Comment:
Nit: I'd be surprised that `code` constructs the code when calling. Maybe rename to `to_code` or can we do the joining more eagerly when constructing `EmbeddedFile`?